File: app/src/main/java/com/pims/medretailers/OrderItem.java
package com.pims.medretailers;

public class OrderItem {
    private boolean isChecked;
    private String rowName = ""; // Internal ERPNext ID for the child row
    private String itemCode = "";
    private String deliveryDate = "";
    private int quantity = 1;
    private double rate = 0.0;
    private String itemName;
    private String date;

    private String uom = "Box";
    private String warehouse = "PIMS MAIN - PE";
    private String notes = "";

    public OrderItem() {} // Constructor

    public String getRowName() { return rowName; }
    public void setRowName(String name) { this.rowName = name; }

    public boolean isChecked() { return isChecked; }
    public void setChecked(boolean checked) { isChecked = checked; }

    public String getItemCode() { return itemCode; }
    public void setItemCode(String itemCode) { this.itemCode = itemCode; }

    public String getDeliveryDate() { return deliveryDate; }
    public void setDeliveryDate(String date) { this.deliveryDate = date; }

    public int getQuantity() { return quantity; }
    public void setQuantity(int quantity) { this.quantity = quantity; }

    public double getRate() { return rate; }
    public void setRate(double rate) { this.rate = rate; }

    public String getItemName() {return itemName; }
    public void setItemName(String itemName) {this.itemName = itemName;}

    // Auto-calculate amount
    public double getAmount() { return quantity * rate; }

    public String getDate() {return date;}
    public void setDate(String date) {
        this.date = date;
    }

    public String getUom() { return uom; }
    public void setUom(String uom) { this.uom = uom; }

    public String getWarehouse() { return warehouse; }
    public void setWarehouse(String warehouse) { this. warehouse = warehouse; }

    public String getNotes() { return notes; }
    public void setNotes(String notes) { this.notes = notes; }



}

File: lib/models/order_item.dart
class OrderItem {
  bool isChecked;
  String itemCode;
  String deliveryDate;
  int quantity;
  double rate;
  String? itemName;
  String? date;
  String uom;
  String warehouse;
  String notes;

  OrderItem({
    this.isChecked = false,
    this.itemCode = "",
    this.deliveryDate = "",
    this.quantity = 0,
    this.rate = 0.0,
    this.itemName,
    this.date,
    this.uom = "Box",
    this.warehouse = "PIMS MAIN - PE",
    this.notes = "",
  });

  double get amount => quantity * rate;

  factory OrderItem.fromJson(Map<String, dynamic> json) {
    return OrderItem(
      itemCode: json['item_code'] ?? "",
      quantity: (json['qty'] ?? 0).toInt(),
      rate: (json['rate'] ?? 0.0).toDouble(),
      uom: json['uom'] ?? "Box",
      warehouse: json['warehouse'] ?? "PIMS MAIN - PE",
      deliveryDate: json['delivery_date'] ?? "",
      notes: json['description'] ?? "",
      itemName: json['item_name'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'item_code': itemCode,
      'qty': quantity,
      'rate': rate,
      'uom': uom,
      'warehouse': warehouse,
      'delivery_date': deliveryDate,
      'description': notes,
    };
  }
}
